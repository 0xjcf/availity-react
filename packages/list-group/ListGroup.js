import React from 'react';
import PropTypes from 'prop-types';
import { ListGroup as RsListGroup } from 'reactstrap';

const ListGroup = ({ cards, selectable, className, ...props }) => {
  const classname = [
    className,
    cards && 'list-group-cards',
    selectable && 'list-group-selectable',
  ]
    .filter(a => a)
    .join(' ');
  return <RsListGroup {...props} className={classname} />;
};

ListGroup.propTypes = {
  cards: PropTypes.bool,
  selectable: PropTypes.bool,
  className: PropTypes.string,
};

export default ListGroup;
