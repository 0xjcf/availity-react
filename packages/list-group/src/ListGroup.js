import React from 'react';
import PropTypes from 'prop-types';
import { ListGroup as RsListGroup } from 'reactstrap';
import classNames from 'classnames';

const ListGroup = React.forwardRef(({ cards, selectable, className, ...props }, ref) => {
  const classes = classNames(className, { 'list-group-cards': cards, 'list-group-selectable': selectable });

  return <RsListGroup ref={ref} {...props} className={classes} />;
});

ListGroup.propTypes = {
  /** Triggers the items (children) to appear as cards. */
  cards: PropTypes.bool,
  /** Triggers the items (children) to appear as selectable when hovered over. */
  selectable: PropTypes.bool,
  /** The name for your class */
  className: PropTypes.string,
};

ListGroup.defaultProps = {
  cards: false,
  selectable: false,
};

export default ListGroup;
