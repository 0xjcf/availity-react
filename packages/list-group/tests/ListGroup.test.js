import React from 'react';
import renderer from 'react-test-renderer';
import ListGroup from '..';

describe('ListGroup', () => {
  test('should render', () => {
    const component = renderer.create(<ListGroup />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render cards', () => {
    const component = renderer.create(<ListGroup cards />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render selectable', () => {
    const component = renderer.create(<ListGroup selectable />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render selectable cards', () => {
    const component = renderer.create(<ListGroup selectable cards />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
