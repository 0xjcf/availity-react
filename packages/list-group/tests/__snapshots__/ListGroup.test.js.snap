// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup should render 1`] = `
<ul
  className="list-group"
/>
`;

exports[`ListGroup should render cards 1`] = `
<ul
  className="list-group-cards list-group"
/>
`;

exports[`ListGroup should render selectable 1`] = `
<ul
  className="list-group-selectable list-group"
/>
`;

exports[`ListGroup should render selectable cards 1`] = `
<ul
  className="list-group-cards list-group-selectable list-group"
/>
`;
